src/irr/asset/CImageLoaderPNG.cpp

```diff
@@ -236,18 +239,32 @@ asset::SAssetBundle CImageLoaderPng::loadAsset(io::IReadFile* _file, const asset
 	}
 
 	// Create array of pointers to rows in image data
-	RowPointers = new png_bytep[Height];
+    uint8_t** RowPointers = _IRR_NEW_ARRAY(png_bytep, Height);;
 	if (!RowPointers)
 	{
 		os::Printer::log("LOAD PNG: Internal PNG create row pointers failure\n", _file->getFileName().c_str(), ELL_ERROR);
 		png_destroy_read_struct(&png_ptr, nullptr, nullptr);
-		image->drop();
         return {};
 	}
 
+    auto texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(image->getImageDataSizeInBytes());
+    auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1u);
+    ICPUImage::SBufferCopy& region = regions->front();
+    //region.imageSubresource.aspectMask = ...; //waits for Vulkan
+    region.imageSubresource.mipLevel = 0u;
+    region.imageSubresource.baseArrayLayer = 0u;
+    region.imageSubresource.layerCount = 1u;
+    region.bufferOffset = 0u;
+    region.bufferRowLength = 0u;//??
+    region.bufferImageHeight = 0u; //??
```

Comment:
see how address is calculated in VK, if this is 0 then its assumed to be tightly packed.